### PR TITLE
Vegalite working example

### PR DESCRIPTION
**Pull request by amitkaps**
Added vegalite.py and related files to get vega-lite working in the
notebook.

### CODE REVIEW

**Comment by domoritz** [comment]
Awesome! 

A few small things before I merge:
* [ ] Please use a dictionary instead of a json string. Or even better, serialize if it is a dict, pass if it is a string. 
* [ ] Can we also change the height of the iframe depending on the size of the chart?
* [ ] Please use a submodule for vl like I did for polestar and voyager.

**Comment by amitkaps** [comment]
Thanks. Will try and fix 2nd and 3rd point. I am not sure I get the 1st point. I am just taking the data output of tools.py and using it as the input in the webpage. Could you clarify?

File: vega/vegalite.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    def __init__(self, columns, data, spec={}):
+        """Constructor
+
+        Args:
+            columns: a list of column names
+            data: list of rows
+            spec: the initial vega-lite spec as a python dictionary"""
+
+        super(Vegalite, self).__init__(columns, data, JS, CSS, TEMPLATE)
+        self.spec = spec
```
Comment (domoritz) [inline vega/vegalite.py]: Convert a dict to a json string here.

**Comment by domoritz** [comment]
> Thanks. Will try and fix 2nd and 3rd point. I am not sure I get the 1st point. I am just taking the data output of tools.py and using it as the input in the webpage. Could you clarify?

I commented on the line in question above. I suggest that you add a new method (`__json_spec`) that always returns a spec as a json string.

**Comment by domoritz** [comment]
@amitkaps Do you want to continue working on this? This looks promising.

**Comment by amitkaps** [comment]
Yes, yes, I will be... Just got sidelined by some client workshops I am doing. I understood you comment on `(_json.spec)` in tools.py, though it will be take me some time to figure it out. I should have the updated merge request by middle of next week. Apologies for being non communicative earlier.

**Comment by amitkaps** [comment]
@domoritz  One other reflection is that in Polestar and Voyager, we are loading a GUI app and hence every time we run it, we need to load all the libraries involved. 

But Vegalite is much simpler and it may be used multiple times in an iPython notebook to explore data. Hence it should be better to load the libraries once into the notebook using something like `output notebook()` and then just pass it HTML + JS to run, without an iFrame.

In our hacking dev sprint, we used require.js to do so once and then just pass plain HTML + JS. See [notebook example here](https://github.com/nischalhp/python-devsprint-viz/blob/master/Vega%2C!Las.ipynb) or the relevant code below.

Do you think we should use this lighter method for Vegalite (pun intended!)?

```
from IPython.display import HTML

%%javascript
require.config({
    shim: {
        d3: { exports: "d3" },
        topojson: {exports: "topojson"},
        vega: {exports: "vega", deps: ["d3", "topojson"] },
        vegalite: {exports: "vegalite", deps: ["vega"] }
        
    },
    paths: {
        d3: "/files/libs/d3.3.5.6.min",
        topojson : "/files/libs/topojson",
        vega: "/files/libs/vega.1.5",
        vegalite: "/files/libs/vegalite"
    }
});

require(["d3","topojson", "vega", "vegalite"], function(d3, topojson,vega, vegalite) {
    console.log(d3.version);
    window.topojson = topojson;
    console.log(topojson.version);
    window.vega = vega;
    console.log(vega.version);
    window.vl = vegalite;
    console.log(vegalite.version);
});

html = """<div id='vis'></div>"""
js = """<script type="text/Javascript">
(function () {
  var vgspec = vl.compile(%s);
  vega.parse.spec(vgspec, function(chart){
  chart({el:"#vis"}).update();
  });
})();
</script>""" % spec
HTML(html+js)
```

**Comment by domoritz** [comment]
I agree that we don't want to reload the whole app just because the data changes. However, I don't think the requirejs code from above fixes that issue. I think we should pass updates into the iframe using post message and then have some code in the iframe that updates the data in polestar or re-renders vega-lite. However, I don't yet know how to inject only the new data into a running cell. 

I honestly don't know what the right solution is and would be happy to look at suggestions.

**Comment by domoritz** [comment]
Closing because of inactivity. But there are some excellent ideas in the pr.

**Pull request closed without merging**